### PR TITLE
guest_os_booting: add new case for ovmf smm

### PR DESCRIPTION
**Pull request by meinaLi**
This PR automate:
VIRT-297068 - [OVMF] Boot vm with ovmf smm

### CODE REVIEW

**Comment by meinaLi** [comment]
```
# avocado run --vt-type libvirt --test-runner=runner guest_os_booting.ovmf_smm
JOB ID     : a2e35fe52781a3886d53d3bc1d4813bd5b23717d
JOB LOG    : /var/lib/avocado/job-results/job-2023-06-01T02.36-a2e35fe/job.log
 (1/5) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_smm.positive_test.tseg_48M: PASS (79.42 s)
 (2/5) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_smm.positive_test.tseg_0M: PASS (83.11 s)
 (3/5) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_smm.positive_test.smm_off: PASS (65.46 s)
 (4/5) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_smm.negative_test.tseg_12345M: PASS (18.24 s)
 (5/5) type_specific.io-github-autotest-libvirt.guest_os_booting.ovmf_smm.negative_test.tseg_-2M: PASS (17.30 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-06-01T02.36-a2e35fe/results.html
JOB TIME   : 265.27 s
```

**Comment by meinaLi** [comment]
The /provider/guest_os_booting/guest_os_booting_base.py used in this PR was submitted in https://github.com/autotest/tp-libvirt/pull/4898 which is not merged now.

**Comment by meinaLi** [comment]
@Yingshun @dzhengfy Please also help to review this PR, thanks.

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_smm.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    vm = env.get_vm(vm_name)
+    vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+    bkxml = vmxml.copy()
+
+    try:
+        try:
+            guest_os.prepare_smm_xml(vm_name, smm_state, smm_tseg_size)
+        except xcepts.LibvirtXMLError as details:
+            if error_msg in str(details):
```
Comment (Yingshun) [inline libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_smm.py:38]: There is no error_msg for positive tests, please update to test the value before checking the exception message.

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_smm.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                test.log.info("Can get expected error message: %s." % error_msg)
+                return
+            else:
+                test.fail("Failed to define the guest %s" % vm_name)
```
Comment (Yingshun) [inline libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_smm.py:42]: For further debug, it's better to show 'details' in the exception. what do you think?

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_smm.py (hunk 1 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_smm.py]: Updated.

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_smm.py (hunk 2 above)
Comment (meinaLi) [reply libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_smm.py]: Yes, using the details to debug is useful. I've already updated it.

File: libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_smm.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    vm = env.get_vm(vm_name)
+    vmxml = vm_xml.VMXML.new_from_inactive_dumpxml(vm_name)
+    bkxml = vmxml.copy()
+
+    try:
+        try:
+            guest_os.prepare_smm_xml(vm_name, smm_state, smm_tseg_size)
+        except xcepts.LibvirtXMLError as details:
+            if error_msg and error_msg in str(details):
+                test.log.info("Can get expected error message: %s." % error_msg)
```
Comment (Yingshun) [inline libvirt/tests/src/guest_os_booting/ovmf_firmware/ovmf_smm.py:39]:
```suggestion
                test.log.info("Can get expected error message: %s.", error_msg)
```

**Review by Yingshun — APPROVED** [review]
LGTM

**Review by dzhengfy — APPROVED** [review]
LGTM

**Comment by dzhengfy** [comment]
@meinaLi Could you fix the CI failure?

**Comment by meinaLi** [comment]
> @meinaLi Could you fix the CI failure?
Please help to review https://github.com/autotest/tp-libvirt/pull/5026. I think this PR is related to this CI failure.

**Comment by dzhengfy** [comment]
Depend on https://github.com/autotest/tp-libvirt/pull/5026

**Pull request merged**